packages/server/src/http/resolveHTTPResponse.ts

```diff
@@ -319,42 +367,40 @@ export async function resolveHTTPResponse<
        * - return void
        */
 
-      try {
-        const headResponse = initResponse({
-          ctx,
-          paths,
-          type,
-          responseMeta: opts.responseMeta,
-        });
-        onHead(headResponse);
-
-        const indexedPromises = new Map(
-          promises.map((promise, index) => [
-            index,
-            promise.then((r) => [index, r] as const),
-          ]),
+      const headResponse = initResponse({
+        ctx,
+        paths,
+        type,
+        responseMeta: opts.responseMeta,
+      });
+      onHead(headResponse);
+
+      const indexedPromises = new Map(
+        promises.map((promise, index) => [
+          index,
+          promise.then((r) => [index, r] as const),
+        ]),
+      );
+      for (let i = 0; i < paths.length; i++) {
+        const [index, untransformedJSON] = await Promise.race(
+          indexedPromises.values(),
         );
-        for (let i = 0; i < paths.length; i++) {
-          const [index, untransformedJSON] = await Promise.race(
-            indexedPromises.values(),
-          );
-          indexedPromises.delete(index);
+        indexedPromises.delete(index);
```

Comment:
We know this can't fail since all procedure calls are wrapped inside a try/catch themselves